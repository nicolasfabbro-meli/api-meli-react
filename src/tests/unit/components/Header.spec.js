import React from 'react';
import Component from '../../../components/Header';

describe('Tag', () => {
  it('Should match snapshot', () => {
    const props = {
      handleSearch: jest.fn(),
    }
    const component = shallow(<Component {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('Should set search terms and call handleSearch when the search icon is clicked', () => {
    const props = {
      handleSearch: jest.fn(),
    }
    const component = shallow(<Component {...props} />);
    const instance = component.instance();
    instance.searchInput = {
      value: '__TEST__',
    };
    const event = {
      type: 'click',
      persist: jest.fn(),
    }
    instance.setSearchTerms(event);
    expect(component.state().searchTerms).toEqual('__TEST__');
    expect(props.handleSearch).toHaveBeenCalled();
    expect(props.handleSearch).toHaveBeenCalledTimes(1);
    expect(props.handleSearch).toHaveBeenCalledWith('__TEST__');
  });

  it('Should set search terms and call handleSearch when enter key is pressed', () => {
    const props = {
      handleSearch: jest.fn(),
    }
    const component = shallow(<Component {...props} />);
    const instance = component.instance();
    instance.searchInput = {
      value: '__TEST__',
    };
    const event = {
      type: 'keydown',
      keyCode: 13,
      persist: jest.fn(),
    }
    instance.setSearchTerms(event);
    expect(component.state().searchTerms).toEqual('__TEST__');
    expect(props.handleSearch).toHaveBeenCalled();
    expect(props.handleSearch).toHaveBeenCalledTimes(1);
    expect(props.handleSearch).toHaveBeenCalledWith('__TEST__');
  });

  it('Should set search terms and not call handleSearch when enter any key (not enter) is pressed', () => {
    const props = {
      handleSearch: jest.fn(),
    }
    const component = shallow(<Component {...props} />);
    const instance = component.instance();
    instance.searchInput = {
      value: '__TEST__',
    };
    const event = {
      type: 'keydown',
      keyCode: 9,
      persist: jest.fn(),
    }
    instance.setSearchTerms(event);
    expect(component.state().searchTerms).toEqual('__TEST__');
    expect(props.handleSearch).not.toHaveBeenCalled();
    expect(props.handleSearch).toHaveBeenCalledTimes(0);
  });
});
