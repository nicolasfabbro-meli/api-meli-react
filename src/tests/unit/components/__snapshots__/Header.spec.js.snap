// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Should match snapshot 1`] = `
"<header>
  <div className='header'>
    <div className='header__left'>
      <a href='https://mercadolibre.com.ar' className='header__link' target='_blank' rel='noopener noreferrer'>
        <img src='logo__large.png' alt='Mercadolibre' className='header__logo' />
      </a>
    </div>
    <div className='header__center'>
      <input type='text' placeholder='Buscar productos, marcas y mas...' className='header__input' onKeyDown={[Function]} />
      <div className='v-divider' />
      <span className='header__icon-search' role='button' onClick={[Function]}>
        <ion-icon name='search-outline' />
      </span>
    </div>
    <div className='header__right' />
  </div>
</header>"
`;
